src/go/configgenerator/route_generator.go

```diff
@@ -272,5 +273,23 @@ func makeHttpRouteMatcher(httpRule *commonpb.Pattern) *routepb.RouteMatch {
 			},
 		},
 	}
-	return &routeMatcher
+	return &routeMatcher, nil
+}
+
+func validateRegexProgramSize(regex string, maxProgramSize int) error {
```

Comment:
should we move this to utils? Nothing routing specific here.

Reply:
Before, I regarded that as a helper function used for route_generator only since most likely it wouldn't be shared by other places. 
I think either way is fine. Move it to `regex.go` now.